src/java.base/unix/classes/sun/nio/fs/UnixCopyFile.java

```diff
@@ -248,11 +269,23 @@ private static void copyFile(UnixPath source,
             // set to true when file and attributes copied
             boolean complete = false;
             try {
+                long transferSize = TRANSFER_SIZE;
+                try {
+                    long bss = Files.getFileStore(source).getBlockSize();
+                    long bst = Files.getFileStore(target).getBlockSize();
+                    if (bss > 0 && bst > 0) {
```

Comment:
This should be UnixFileStoreAttributes.get(source).blockSize() to avoid calling from the provider to the API or needing to create a FileStore.

Reply:
Fixed in bc4096c6cd4f1a060a8cacb154ac090c98f94aa3.